tgui/packages/tgui/interfaces/MinimapController.js

```diff
@@ -0,0 +1,242 @@
+                          options={placable_marker_states}
+                          onSelected={(value) => setImage(value)}
+                          mt="10px"
+                        />
+                        <Flex mt="10px" justify="space-between">
+                          <Flex.Item>
+                            <NumberInput
+                              width="70px"
+                              minValue={1}
+                              maxValue={300}
+                              value={x}
+                              format={value => "x, " + value}
+                              onDrag={(e, value) => setX(value)}
+                            />
+                          </Flex.Item>
+                          <Flex.Item>
+                            <NumberInput
+                              width="70px"
+                              minValue={1}
+                              maxValue={300}
+                              value={y}
+                              format={value => "y, " + value}
+                              onDrag={(e, value) => setY(value)}
+                            />
+                          </Flex.Item>
+                        </Flex>
+                        <Button
+                          fluid
+                          textAlign="center"
+                          color={selecting_coordinates ? "orange" : "default"}
+                          content={selecting_coordinates ? "Select Position" : "Select (x, y) From Map"}
+                          onClick={() => act('location_from_minimap')}
+                          mt="10px"
+                        />
+                        <Flex mt="20px" justify="space-between">
+                          <Flex.Item>
+                            <Button
+                              icon="check"
+                              color="green"
+                              content="Confirm"
+                              onClick={() => newMarker()}
+                            />
+                          </Flex.Item>
+                          <Flex.Item>
+                            <Button
+                              icon="xmark"
+                              color="red"
+                              content="Cancel"
+                              onClick={() => toggleNewMarkerMenu()}
+                            />
+                          </Flex.Item>
+                        </Flex>
+                      </Flex.Item>
+                    </Flex>
+                  </Box>
+                </Modal>
+              )}
+              <Section scrollable fill>
+                <Flex direction="column">
+                  <Flex.Item>
+                    <Flex.Item>
+                      {Object.keys(minimap_markers).map(marker => (
+                        <Flex key={data.minimap_markers[marker]}
+                          backgroundColor={"rgba(0, 0, 0, 0.3)"}
+                          justify="space-between"
```

Comment:
CSS classes for iterated objects so we're not repeating inline styles (i think?)